### ISSUE TITLE
STM32F407ZGT6 / STM32F042K6: unknown chip id - RESET 0x5fa0004

### ISSUE THREAD

**Issue by goog** [texane/stlink#791]
When submitting a feature request, try to reuse the list and add/remove what is appropriate.
 Place a `X` between the brackets `[X]` to mark the list item.

- [X] Programmer/board type: Stlink/v2
- [x] Programmer firmware version: e.g STSW-LINK007 2.27.15
- [X] Operating system: Linux
- [X] Stlink tools version and/or git commit hash: v1.5.1-21-gc6836b4
- [X] Stlink commandline tool name: `st-flash`

ubuntu12.04 64bit STM32F407ZGT6

A as-detailed description possible of the problem with debug output when available.

Output:

```
cheng@ubuntu:~/stlink/build/Release$ ./st-flash write /home/cheng/stm32f4/arm-rtems4.11/c/stm32f4/testsuites/samples/hello/hello.bin 0x8000000
st-flash 1.5.1-21-gc6836b4
2019-04-07T23:22:21 INFO common.c: Loading device parameters....
2019-04-07T23:22:21 WARN common.c: unknown chip id! 0x5fa0004
```

**Comment by set321go** [comment]
I'm seeing the same issue when trying to program STM32F042K6, which was talking to `st-info --chipid` initially. I do notice that that `0x5fa0004` is the reset command.

**Comment by Nightwalker-87** [comment]
@goog: Please verify if the problem still exists in Release v1.6.0.

**Comment by Nightwalker-87** [comment]
Resolved by either #715 (swapped SWDIO and SWCLK lines) or #832 (firmware).
